### PR TITLE
feat: add bypass_cache_on_request_headers to cdn_policy

### PR DESCRIPTION
**Pull request by hhollenstain**
Add bypass_cache_on_request_headers to cdn_policy. This was added back on compute_backend on the provider on [4.64.0](https://github.com/hashicorp/terraform-provider-google/blob/main/CHANGELOG.md#4640-may-8-2023)

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -274,6 +274,13 @@ resource "google_compute_backend_service" "default" {
           include_named_cookies  = each.value.cdn_policy.cache_key_policy.include_named_cookies
         }
       }
+
+      dynamic "bypass_cache_on_request_headers" {
+        for_each = toset(each.value.cdn_policy.bypass_cache_on_request_headers) != null ? [1] : []
```
Comment (imrannayer) [inline main.tf:279]: @hhollenstain Is this logic correct? Assuming you want to create multiple blocks of `bypass_cache_on_request_headers` depending on how many values r passed in your list. I think logic should be as follows:

```
for_each = toset(each.value.cdn_policy.bypass_cache_on_request_headers) != null ? each.value.cdn_policy.bypass_cache_on_request_headers : []
```

**Comment by imrannayer** [comment]
@hhollenstain thanks for the PR. Can you plz check the logic?

File: main.tf (hunk 1 above)
Comment (hhollenstain) [reply main.tf]: Yep, you are correct. I will fix it up.

**Comment by imrannayer** [comment]
@hhollenstain test is failing:

Code:

```
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:   # module.gce-lb-http.google_compute_backend_service.default["default"] will be created
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:   + resource "google_compute_backend_service" "default" {
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:       + connection_draining_timeout_sec = 300
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:       + creation_timestamp              = (known after apply)
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:       + enable_cdn                      = true
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:       + fingerprint                     = (known after apply)
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:       + generated_id                    = (known after apply)
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:       + health_checks                   = (known after apply)
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:       + id                              = (known after apply)
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:       + load_balancing_scheme           = "EXTERNAL"
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:       + name                            = "mig-http-lb-backend-default"
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:       + port_name                       = "http"
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:       + project                         = "ci-int-lb-http-368b"
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:       + protocol                        = "HTTP"
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:       + self_link                       = (known after apply)
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:       + session_affinity                = (known after apply)
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:       + timeout_sec                     = 10
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185: 
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:       + backend {
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + balancing_mode               = "UTILIZATION"
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + capacity_scaler              = 1
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + group                        = (known after apply)
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + max_connections              = (known after apply)
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + max_connections_per_endpoint = (known after apply)
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + max_connections_per_instance = (known after apply)
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + max_rate                     = (known after apply)
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + max_rate_per_endpoint        = (known after apply)
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + max_rate_per_instance        = (known after apply)
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + max_utilization              = (known after apply)
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:         }
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185: 
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:       + cdn_policy {
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + cache_mode                   = "CACHE_ALL_STATIC"
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + client_ttl                   = 7200
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + default_ttl                  = 3600
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + max_ttl                      = 10800
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + negative_caching             = (known after apply)
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + serve_while_stale            = (known after apply)
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + signed_url_cache_max_age_sec = 3600
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185: 
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + bypass_cache_on_request_headers {
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:               + header_name = "default"
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:             }
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + bypass_cache_on_request_headers {
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:               + header_name = "default"
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:             }
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185: 
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:           + cache_key_policy {
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:               + include_host          = true
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:               + include_named_cookies = [
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:                   + "__next_preview_data",
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:                   + "__prerender_bypass",
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:                 ]
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:               + include_protocol      = true
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:               + include_query_string  = true
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:             }
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:         }
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185:     }
TestAll/examples/cdn-policy 2023-11-16T14:54:03Z command.go:185: 
```


Error:

```
TestAll/examples/cdn-policy 2023-11-16T14:55:34Z command.go:185: 
TestAll/examples/cdn-policy 2023-11-16T14:55:34Z command.go:185: Error: Error creating BackendService: googleapi: Error 400: Invalid value for field 'resource.cdnPolicy.bypassCacheOnRequestHeaders[1].headerName': 'default'. Cannot have multiple headers with the same name: default, invalid
TestAll/examples/cdn-policy 2023-11-16T14:55:34Z command.go:185: 
TestAll/examples/cdn-policy 2023-11-16T14:55:34Z command.go:185:   with module.gce-lb-http.google_compute_backend_service.default["default"],
TestAll/examples/cdn-policy 2023-11-16T14:55:34Z command.go:185:   on ../../main.tf line 182, in resource "google_compute_backend_service" "default":
TestAll/examples/cdn-policy 2023-11-16T14:55:34Z command.go:185:  182: resource "google_compute_backend_service" "default" {
TestAll/examples/cdn-policy 2023-11-16T14:55:34Z command.go:185: 
TestAll/examples/cdn-policy 2023-11-16T14:55:34Z retry.go:99: Returning due to fatal error: FatalError{Underlying: error while running command: exit status 1; 
Error: Error creating BackendService: googleapi: Error 400: Invalid value for field 'resource.cdnPolicy.bypassCacheOnRequestHeaders[1].headerName': 'default'. Cannot have multiple headers with the same name: default, invalid

  with module.gce-lb-http.google_compute_backend_service.default["default"],
  on ../../main.tf line 182, in resource "google_compute_backend_service" "default":
 182: resource "google_compute_backend_service" "default" {
}
    apply.go:34: 
        	Error Trace:	/builder/home/go/pkg/mod/github.com/gruntwork-io/terratest@v0.43.13/modules/terraform/apply.go:34
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.9.1/pkg/tft/terraform.go:437
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.9.1/pkg/tft/terraform.go:447
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.9.1/pkg/tft/terraform.go:471
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.9.1/pkg/utils/stages.go:31
        	            				/builder/home/go/pkg/mod/github.com/!google!cloud!platform/cloud-foundation-toolkit/infra/blueprint-test@v0.9.1/pkg/tft/terraform.go:471
        	Error:      	Received unexpected error:
        	            	FatalError{Underlying: error while running command: exit status 1; 
        	            	Error: Error creating BackendService: googleapi: Error 400: Invalid value for field 'resource.cdnPolicy.bypassCacheOnRequestHeaders[1].headerName': 'default'. Cannot have multiple headers with the same name: default, invalid
        	            	
        	            	  with module.gce-lb-http.google_compute_backend_service.default["default"],
        	            	  on ../../main.tf line 182, in resource "google_compute_backend_service" "default":
        	            	 182: resource "google_compute_backend_service" "default" {
        	            	}
        	Test:       	TestAll/examples/cdn-policy
2023/11/16 14:55:34 RUN_STAGE env var set to apply
2023/11/16 14:55:34 Skipping stage teardown
--- FAIL: TestAll (96.98s)
```

**Comment by hhollenstain** [comment]
@imrannayer Thanks for getting me the test results 

I hooked this up to a env to make sure this is rendering as expected and needed to act on proper value. Just pushed it up and seems like you are already on it. Thanks for all the help.

**Pull request merged**